### ISSUE TITLE
Framework Integrations other than React?

### ISSUE THREAD

**Issue by XiNiHa** [daangn/stackflow#277]
I'd like to use Stackflow with other frameworks than React, especially with SolidJS. While I'm somewhat interested in building the integration by myself, it looks not that easy to start with since the current integration code for React looks somewhat large to just read it without any docs, and the docs related to the architecture of the package are marked as in construction. Is it possible to get some support for SolidJS integration, or at least the docs to be prepared anytime soon?

**Comment by XiNiHa** [comment]
I've spent some time trying to create SolidJS integration for Stackflow and realized that all the workspace settings and extensions are optimized for React. I have no idea how to manage extensions for different framework integrations available. Do you have any good ideas for doing this? Do you think that the plugins should be developed for each framework integration again and again?

**Comment by tonyfromundefined** [comment]
I agree with what you pointed out! Soon we will be moving some plug-in interface implementations incubated in integration/react to "core". (The `StackflowPlugin` interface is now a common core interface, but the implementation is only in react integration.🥲 It not makes sense. 😂)

For this, several steps are required:

- First, we will move the entire state management (including event dispatching) logic to the core module. (currently in react integration's `CoreProvider`)
- Second, we will expose lifecycle hooks in our core state management logic. (Common plug-in interfaces can be inserted.)
- Thirdly (if possible), we are going to use `React.useSyncExternalStore()` instead of `useState()` in React integration. (We will drop support for React 17 and lower later)

**Comment by XiNiHa** [comment]
Sounds great! It'd be great for me to have further works linked with this issue, so I can get some notifications from it! 🙏

**Comment by tonyfromundefined** [comment]
Currently, Karrot is undergoing migration, so stability and fulfilling existing requirements are of high priority. Perhaps an improvement schedule will be set at a later date, but I will let you know. Thanks for your contribution! 🤩

**Comment by gron1gh1** [comment]
I had the same concern about the Vue integration. I want to get a notification, too!

**Comment by XiNiHa** [comment]
I recently started my new attempt to make an integration for Solid. While mostly going well, I have these concerns/questions:
- Would it make sense to provide a granular way to subscribe state changes in the core? Currently, it's forced to run [reconcile](https://docs.solidjs.com/reference/store-utilities/reconcile) to make the state update fine-grained. While this works for now, it'd be better in performance perspective if there's a finer-grained interface I can tap on to bail out of reconcilation.
- Since UI and renderer plugins are tied to UI frameworks, the current approach I'm taking is to create duplicated packages for each of the first-party UI/renderer plugins (`plugin-renderer-basic-solid`, `plugin-basic-ui-solid`, ...) However, this resulted in lots of duplicated code, increasing maintenance costs. To minimize this, I have some options in mind. What would be the best way to handle the problem?
  - Extract only the VE styles from `basic-ui` as a separate package, and import them from each implementation
  - Extract all the styles and logics as a common interface, separated as a package, then hook them from each implementation
  - Same as above, but don't split the package and provide implementations for each framework as subpath exports. (maybe the React implementation should also be exported from `.` to avoid breakages... or should we just make a major version bump?) Make UI frameworks as optional peer dependencies
While I definitely want to upstream the changes, I'm open to maintain a separate fork if you're worried of maintenance cost.

**Comment by Moon-DaeSeung** [comment]
@XiNiHa 

If your requirements are not accepted, what do you think about ssgoi.dev ↗?
I’m currently working on implementing SolidJS support there as well.

**Comment by XiNiHa** [comment]
That looks promising! I'll be looking forward to how it'll evolve further 👀
